doc/security/implementation_guidelines/hardware/index.md

```diff
@@ -0,0 +1,409 @@
+    where the inputs are particularly sensitive (requests to unlock, privilege
+    increase requests, debug mode enables, etc).
+
+### **Recommendation 7**: DFT issues
+
+1.  Entry and exit from scan mode should cause a reset to prevent insertion or
+    exfiltration of sensitive values
+2.  Ensure that when in production (e.g. not in lab debug) environments, scan
+    chains are disabled
+3.  Processor debug paths (via JTAG) may need to be disabled in production modes
+4.  Beware of self-repair or redundant-row/columns schemes for memories (SRAM
+    and OTP), as they can be exploited to misdirect reads to
+    adversary-controlled locations
+
+### **Recommendation 8**: Power management issues
+
+1.  If module is not in an always-on power domain, consider that a sleep/wake
+    sequence can be used to force a re-derivation of secrets needed in the
+    module, as many times as desired by the adversary
+2.  Fine-grained clock gating should never be used for any module that processes
+    secret data, only coarse-grained (module-level) gating is acceptable. (Fine
+    grained gates essentially compute `clock_gate = D ^ Q` which often acts as
+    an SCA "amplifier").
+
+### **Recommendation 9**: "Synchronization" (consistency) issues
+
+1.  If a module interacts with other modules in a stateful way (think of two
+    data-transfer counters moving in ~lockstep, but the counts are not sent back
+    and forth for performance optimization), what happens if:
+    1.  One side is reset and the other is not
+    2.  One side is clock-gated and the other is not
+    3.  One side is power-gated and the other is not
+    4.  The counter on one side is glitched
+2.  Generally these kind of blind lockstep situations should be avoided where
+    possible, and current module/interface status should exchanged in both
+    directions and constantly checked for validity/consistency
+
+### **Recommendation 10**: Recovery mechanism considerations
+
+1.  What happens if a security mechanism fails? (Classic problem of this variety
+    is on-die sensors being too sensitive and resetting the chip) Traditionally,
+    fuses can disable some mechanisms if they are faulty.
+2.  Could an adversary exploit a recovery mechanism? (If a sensor can be
+    fuse-disabled, wouldn't the adversary just do that? See 4.4 above.)
+
+### **Recommendation 11**: Optimization concerns
+
+1.  Sometimes synthesis will optimize away redundant (but necessary for
+    security) logic - `dont_touch` or `size_only` attributes may sometimes be
+    needed, or even more aggressive preservation strategies. Example: when using
+    the sparse FSM encoding, use the `prim_flop` component for the state vector
+    register.
+2.  Value-and-complement strategies can also be optimized away, or partially
+    disconnected such that only half of the datapath is contributing to the
+    logic, or a single register with both Q & Qbar outputs becomes the source of
+    both values to save area.
```

Comment:
I agree that this is a concern, but it doesn't come with a recommendation :-) "Squint hard at your backend team", maybe?